sunbird-es-utils/src/main/java/org/sunbird/common/ElasticSearchTcpImpl.java

```diff
@@ -0,0 +1,652 @@
+      promise.success("ERROR");
+      return promise.future();
+    }
+    try {
+      data.put("identifier", identifier);
+      IndexResponse response =
+          ConnectionManager.getClient().prepareIndex(index, _DOC, identifier).setSource(data).get();
+      ProjectLogger.log(
+          "ElasticSearchTcpImpl:save: "
+              + "Save value=="
+              + response.getId()
+              + " "
+              + response.status(),
+          LoggerEnum.INFO.name());
+      ProjectLogger.log(
+          "ElasticSearchTcpImpl:save: method end at =="
+              + System.currentTimeMillis()
+              + " for INdex "
+              + index
+              + " ,Total time elapsed = "
+              + ElasticSearchHelper.calculateEndTime(startTime),
+          LoggerEnum.PERF_LOG);
+      promise.success(response.getId());
+      return promise.future();
+    } catch (Exception e) {
+      ProjectLogger.log(
+          "ElasticSearchTcpImpl:save: Error while saving index " + index + " id : " + identifier,
+          e);
+      ProjectLogger.log(
+          "ElasticSearchTcpImpl:save: method end at =="
+              + System.currentTimeMillis()
+              + " for Index "
+              + index
+              + " ,Total time elapsed = "
+              + ElasticSearchHelper.calculateEndTime(startTime),
+          LoggerEnum.PERF_LOG);
+      promise.failure(e);
+      promise.success("");
+    }
+    return promise.future();
+  }
+
+  /**
+   * This method will provide data form ES based on incoming identifier. we can get data by passing
+   * index and identifier values , or all the three
+   *
+   * @param identifier String
+   * @return Map<String,Object> or null
+   */
+  @Override
+  public Future<Map<String, Object>> getDataByIdentifier(String index, String identifier) {
+    long startTime = System.currentTimeMillis();
+    Promise<Map<String, Object>> promise = Futures.promise();
+    ProjectLogger.log(
+        "ElasticSearchTcpImpl:getDataByIdentifier: method started at =="
+            + startTime
+            + " for index "
+            + index,
+        LoggerEnum.PERF_LOG);
+    GetResponse response = null;
+    if (StringUtils.isBlank(index) || StringUtils.isBlank(identifier)) {
+      ProjectLogger.log(
+          "ElasticSearchTcpImpl:getDataByIdentifier: Invalid request is coming.", LoggerEnum.INFO);
+      promise.success(new HashMap<>());
+      return promise.future();
+    } else {
+      response = ConnectionManager.getClient().prepareGet(index, _DOC, identifier).get();
+    }
+    if (response == null || null == response.getSource()) {
+      promise.success(new HashMap<>());
+      return promise.future();
+    }
+    long elapsedTime = ElasticSearchHelper.calculateEndTime(startTime);
+    ProjectLogger.log(
+        "ElasticSearchTcpImpl:getDataByIdentifier: method "
+            + " for index "
+            + index
+            + " ,Total time elapsed = "
+            + elapsedTime,
```

Comment:
use here  ElasticSearchHelper.calculateEndTime(startTime); directly in place of elapsedTime

Reply:
Done